src/util/namespace.h

```diff
@@ -27,10 +27,21 @@ class union_tag_typet;
 class struct_tag_typet;
 class c_enum_tag_typet;
 
+/// Basic interface for a namespace. This is not used
+/// in practice, as the one being used is \ref namespacet
+/// which uses two symbol tables, and \ref multi_namespacet
+/// which can combine more than two.
 class namespace_baset
 {
 public:
   // service methods
+
+  /// Lookup a symbol in the namespace.
+  /// \param name: The name of the symbol to perform a lookup for.
+  /// \return: A reference to the symbol found.
+  /// \remarks: This assumes that there exists a symbol of that
+  /// name in the namespace. In the event it's not found, an
+  /// INVARIANT is being hit.
```

Comment:
Perhaps better phrased "It is a PRECONDITION that the symbol name exists in the namespace' ?